lib/compress/zstd_compress_internal.h

```diff
@@ -131,6 +131,20 @@ typedef struct {
     U32 lowLimit;           /* below that point, no more valid data */
 } ZSTD_window_t;
 
+typedef struct {
+    U32 offset;
+    U32 litLength;  /* In an LDM, represents size of the literal block preceding the match */
+    U32 matchLength;
+} rawSeq;
+
+typedef struct {
+  rawSeq* seq;              /* The start of the sequences */
+  size_t pos;               /* The position where reading stopped. <= size. */
+  size_t size;              /* The number of sequences. <= capacity. */
+  size_t capacity;          /* The capacity starting from `seq` pointer */
+  size_t bytesRead;         /* The bytes-based analogue to "pos" - read position in bytes rather than idx */
```

Comment:
This increases the size of `rawSeqStore_t`,
which in turns increases the budget needed for any array of this type.

Are you certain that all allocations and size evaluations involving this type are properly resized ?
(i.e. typically use a multiple of `sizeof(rawSeqStore_t)`)